tests/net/lib/coap/src/main.c

```diff
@@ -260,6 +315,15 @@ static int test_parse_empty_pdu(void)
 		goto done;
 	}
 
+	if (coap_header_get_type(&cpkt) == COAP_TYPE_RESET) {
```

Comment:
This test case is to test simple PDU packet, not related to Observer functionality. Do not mix here. Write a separate test case to verify Observer functionality.